packages/api/src/tx/modules/ecocredit/basket/v1/create_amino.ts

```diff
@@ -52,14 +52,14 @@ export function createBasketConverter(): AminoConverter {
         name,
         description,
         exponent: exponent || undefined,
-        disable_auto_retire: Boolean(disableAutoRetire),
+        disable_auto_retire: disableAutoRetire || undefined,
         credit_type_abbrev: creditTypeAbbrev,
         allowed_classes: allowedClasses,
-        date_criteria: dateCriteria && {
-          min_start_date: dateCriteria?.minStartDate,
-          start_date_window: dateCriteria?.startDateWindow,
-          years_in_the_past: dateCriteria?.yearsInThePast,
-        },
+        date_criteria: (dateCriteria && {
+          min_start_date: dateCriteria.minStartDate || undefined,
+          start_date_window: dateCriteria.startDateWindow || undefined,
+          years_in_the_past: dateCriteria.yearsInThePast || undefined,
+        }) || undefined,
```

Comment:
I've updated the amino converter for CreateBasket to render all default values as undefined.

We're going to need to go through every individual amino converter and ensure a similar behavior to this is adhered to.

@amaurym do you have nay thoughts on this? I'm actually wondering now if it would be better to do this conversion of default values all in one place (as part of the cosmjs patch) so that we don't need to be manually preparing all of these amino converters with custom support for default values.

In reality, every field should probably be `foo_field: fooField || undefined`, but since many fields are required and cannot have default values, we simply ignore them in the converters?

Reply:
It seems you're right, cosmjs doesn't strip away fields with golang's default values. But cosmjs also doesn't seem to do `|| undefined`. So is that a cosmjs bug too?

Fine with any of the 2 solutions (`|| undefined` everywhere or do it in a patch), though probably the `|| undefined` is safer since it's happening on a codebase we have more control/knowledge of.